lib/influxdb/client.rb

```diff
@@ -41,118 +46,28 @@ class Client
+      @config = InfluxDB::Config.new(opts)
+      @stopper = false
 
-      @worker = InfluxDB::Worker.new(self) if @async
-      self.udp_client = opts[:udp] ? InfluxDB::UDPClient.new(opts[:udp][:host], opts[:udp][:port]) : nil
-
-      at_exit { stop! }
-    end
-
-    ## allow options, e.g. influxdb.create_database('foo', replicationFactor: 3)
-    def create_database(name, options = {})
-      url = full_url("/db")
-      options[:name] = name
-      data = JSON.generate(options)
-      post(url, data)
-    end
+      @writer = self
 
-    def delete_database(name)
-      delete full_url("/db/#{name}")
-    end
-
-    def get_database_list
-      get full_url("/db")
-    end
-
-    def create_cluster_admin(username, password)
-      url = full_url("/cluster_admins")
-      data = JSON.generate({:name => username, :password => password})
-      post(url, data)
-    end
-
-    def update_cluster_admin(username, password)
-      url = full_url("/cluster_admins/#{username}")
-      data = JSON.generate({:password => password})
-      post(url, data)
-    end
-
-    def delete_cluster_admin(username)
-      delete full_url("/cluster_admins/#{username}")
-    end
-
-    def get_cluster_admin_list
-      get full_url("/cluster_admins")
-    end
-
-    def create_database_user(database, username, password, options={})
-      url = full_url("/db/#{database}/users")
-      data = JSON.generate({:name => username, :password => password}.merge(options))
-      post(url, data)
-    end
-
-    def update_database_user(database, username, options = {})
-      url = full_url("/db/#{database}/users/#{username}")
-      data = JSON.generate(options)
-      post(url, data)
-    end
-
-    def delete_database_user(database, username)
-      delete full_url("/db/#{database}/users/#{username}")
-    end
-
-    def get_database_user_list(database)
-      get full_url("/db/#{database}/users")
-    end
-
-    def get_database_user_info(database, username)
-      get full_url("/db/#{database}/users/#{username}")
-    end
-
-    def alter_database_privilege(database, username, admin=true)
-      update_database_user(database, username, :admin => admin)
-    end
-
-    def continuous_queries(database)
-      get full_url("/db/#{database}/continuous_queries")
-    end
-
-    def get_shard_list()
-      get full_url("/cluster/shards")
-    end
+      if config.async?
+        @writer = InfluxDB::Writer::Async.new(self, config.async)
+      elsif config.udp?
+        @writer = InfluxDB::Writer::UDP.new(self, config.udp)
+      end
 
-    def delete_shard(shard_id, server_ids)
-      data = JSON.generate({"serverIds" => server_ids})
-      delete full_url("/cluster/shards/#{shard_id}"), data
+      at_exit { stop! }
     end
 
-    def write_point(name, data, async=@async, time_precision=@time_precision)
+    # Write data point for series `name`
+    def write_point(name, data)
       data = data.is_a?(Array) ? data : [data]
```

Comment:
`Array(data)` does exactly this. If you pass it an array, it just returns that array, but if you pass it splatted arguments, it returns them as an array. http://ruby-doc.org/core-2.0.0/Kernel.html#method-i-Array